### PR TITLE
chore(cat-voices): Removing code generated files from repo

### LINKED ISSUE #157: Move Dart code generation to CI/CD

### PR DESCRIPTION
**Pull request by LynxLynxx**
# Description

Removes all generated files from catalyst-voices dir except files from public packages located in /packages/libs/

## Related Issue(s)

Resolves #157

## Description of Changes

- [x] Adding files to gitignore
- [x] Changing earthly targets
- [x] Adding a section for code generation in README file

## Related Pull Requests

#1187 

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Comment by minikin** [comment]
Is this one dependent on #1187?

**Review by minikin — CHANGES_REQUESTED** [review]
Can we please update https://github.com/input-output-hk/catalyst-voices/blob/main/catalyst_voices/README.md and add a new section about code generation, including what we use it for, how to use it, etc.?

File: catalyst_voices/README.md [hunk 1]
```diff
@@ -97,6 +97,31 @@ you can use the following command:
...
+To save generated code locally run in root directory:
+
+```sh
+earthly ./catalyst_voices+code-generator --save_locally=true
+```
+
+Keep in mind that You will need GITHUB_TOKEN to be able to run this earthly target.
+In root directory there is a template file `.secret.template` save it as `.secret` and fill in the GITHUB_TOKEN.
+This file should be ignored by git, but make this is the case.
+
```
Comment (minikin) [inline catalyst_voices/README.md:124]:
```suggestion
### Code Generation

This project utilizes automatic code generation for the following components:
- Catalyst Gateway OpenAPI
- Localization files
- Asset files
- Navigation route files

#### Running Code Generation

 **Basic Generation**
 
To generate code, run the following command in the root directory:

`earthly ./catalyst_voices+code-generator`

**Local Saving**

To save the generated code locally, use the `--save_locally` flag:

`earthly ./catalyst_voices+code-generator --save_locally=true`

#### GitHub Token / PAT Setup

 **‼️Important**

A valid `GITHUB_TOKEN`/ `PAT` is required to run the earthly target.

**Token Configuration:**
1. Locate the `.secret.template` file in the root directory
2. Create a copy of this file and name it `.secret`
3. Add your `GITHUB_TOKEN` to the `.secret` file

#### Security Notes
- The `.secret` file should be included in `.gitignore`
- Verify that git does not track the `.secret` file before committing
```

Comment (minikin) [reply catalyst_voices/README.md:124]: @LynxLynxx, Please update the Table of Contents in the README file.

**Review by minikin — CHANGES_REQUESTED** [review]

File: catalyst_voices/Earthfile [hunk 2]
```diff
@@ -20,38 +19,41 @@ builder:
...
     LET local_gen_code_path = packages/internal/catalyst_voices_services/lib/generated/catalyst_gateway/
 
-    FROM +builder
+    RUN melos l10n
+    RUN melos build_runner
+
+    IF [ $save_locally = true ]
+        RUN find . \( -name "*.g.dart" -o -name "*.freezed.dart" -o -name "*.gen.dart" -o -name "catalyst_voices_localizations*.dart" \)
+
+        FOR generated_file IN $(find . \( -name "*.g.dart" -o -name "*.freezed.dart" -o -name "*.gen.dart" -o -name "catalyst_voices_localizations*.dart" \))
```
Comment (dtscalac) [inline catalyst_voices/Earthfile:38]: Make sure these file extensions are synced with the .gitignore here:

Screenshot 2024-11-14 at 10 57 10

File: catalyst_voices/README.md (hunk 1 above)
Comment (LynxLynxx) [reply catalyst_voices/README.md]: Good suggestion for better readability! Thanks!

File: catalyst_voices/Earthfile (hunk 2 above)
Comment (LynxLynxx) [reply catalyst_voices/Earthfile]: Good catch!. Thank you!

**Review by dtscalac — APPROVED** [review]
LGTM

**Review by minikin — APPROVED** [review]
LGTM

**Pull request merged**